packages/builtin/string.pony

```diff
@@ -207,7 +207,13 @@ class val String is (Seq[U8] & Comparable[String box] & Stringable)
     null terminator.
     """
     if _alloc <= len then
-      _alloc = len.min(len.max_value() - 1) + 1
+      let max = len.max_value() - 1
+      let min_alloc = len + 1
+      if min_alloc < (max / 2) then
+        _alloc = min_alloc.next_pow2()
+      else
+        _alloc = min_alloc.min(max)
```

Comment:
I think this still has a problem.  In the case where `len` is equal to `len.max_value() - 1`, the original code would allocate a size of `len.max_value()`, due to the `+ 1` being at the end.  This code snippet should use `let min_alloc = len.min(len.max_value() - 1) + 1` (the calculation in the original snippet).

Also, I think the branch condition should use `<= (max / 2)` instead of `< (max / 2)`, since `(max / 2)` is going to be an odd number.